pkg/ddc/juicefs/datasetinfo_parser.go

```diff
@@ -57,3 +57,28 @@ func parseCacheInfoFromConfigMap(configMap *v1.ConfigMap) (cacheinfo map[string]
 	}
 	return configmapinfo, nil
 }
+
+func GetFSInfoFromConfigMap(client client.Client, name string, namespace string) (info map[string]string, err error) {
+	configMapName := fmt.Sprintf("%s-juicefs-values", name)
+	configMap, err := kubeclient.GetConfigmapByName(client, configMapName, namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "GetConfigMapByName error when GetCacheInfoFromConfigmap")
+	}
+	return parseFSInfoFromConfigMap(configMap)
+}
+
+func parseFSInfoFromConfigMap(configMap *v1.ConfigMap) (info map[string]string, err error) {
+	var value JuiceFS
+	info = map[string]string{}
+	if v, ok := configMap.Data["data"]; ok {
+		if err := yaml.Unmarshal([]byte(v), &value); err != nil {
```

Comment:
Define a new err variable inside the if statement, it will overwrite the outer err variable. I think you need use `err = yaml.Unmarshal([]byte(v), &value)`

Reply:
done